baserun/exporter.py

```diff
@@ -0,0 +1,172 @@
+            if key.startswith(prefix):
+                parts = key[len(prefix) + 1 :].split(".")
+                index, field = int(parts[0]), parts[1]
+                if index not in result:
+                    result[index] = {}
+                result[index][field] = value
+
+        return [v for k, v in sorted(result.items())]
+
+    def export(self, spans: Sequence[ReadableSpan]):
+        from baserun import Baserun
+
+        for span in spans:
+            if span.name.startswith("baserun.parent") or not span.name.startswith(
+                "baserun"
+            ):
+                continue
+
+            status = Status(
+                message=span.status.description, code=span.status.status_code.value
+            )
+            prompt_messages = [
+                Message(**message_attrs)
+                for message_attrs in self._extract_prefix_dicts(
+                    span.attributes, SpanAttributes.LLM_PROMPTS
+                )
+            ]
+            # TODO: Prompt messages for completions input which are just a string
+            completions = [
+                Message(**message_attrs)
+                for message_attrs in self._extract_prefix_dicts(
+                    span.attributes, SpanAttributes.LLM_COMPLETIONS
+                )
+            ]
+
+            # Trace IDs are huge integers, so they must be encoded as bytes
+            trace_id_int = span.context.trace_id
+            trace_id = trace_id_int.to_bytes(
+                (trace_id_int.bit_length() + 7) // 8, "big"
+            )
+
+            vendor = span.attributes.get(SpanAttributes.LLM_VENDOR)
+
+            run_str = span.attributes.get(SpanAttributes.BASERUN_RUN)
+            if not run_str:
+                logger.warning("Baserun run attribute not set, cannot submit run")
+
+            run = Baserun.deserialize_run(run_str)
+
+            span_message = Span(
+                run_id=run.run_id,
+                trace_id=trace_id,
+                span_id=span.context.span_id,
+                name=span.name,
+                vendor=vendor,
+                status=status,
+                total_tokens=span.attributes.get(
+                    SpanAttributes.LLM_USAGE_TOTAL_TOKENS, 0
+                ),
+                completion_tokens=span.attributes.get(
+                    SpanAttributes.LLM_USAGE_COMPLETION_TOKENS, 0
+                ),
+                prompt_tokens=span.attributes.get(
+                    SpanAttributes.LLM_USAGE_PROMPT_TOKENS, 0
+                ),
+                prompt_messages=prompt_messages,
+                model=span.attributes.get(SpanAttributes.LLM_REQUEST_MODEL, ""),
+                completions=completions,
+                request_type=span.attributes.get(SpanAttributes.LLM_REQUEST_TYPE, ""),
+            )
+            span_message.start_time.FromNanoseconds(span.start_time)
+            span_message.end_time.FromNanoseconds(span.end_time)
+
+            if vendor == ANTHROPIC_VENDOR_NAME:
```

Comment:
Bug where anthropic attributes aren't getting exported

Reply:
These are all of the response attributes we get from anthropic- i added stop and max tokens to the `span_message` instantiation since it's (now) the same with both vendors.